graphql/resolvers.go

```diff
@@ -246,7 +246,88 @@ func (r *queryResolver) Projects(ctx context.Context) (*Projects, error) {
 	return &pjs, nil
 }
 
-func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCategory *TaskSortCategory, sortDirection *SortDirection, page *int, limit *int, testName *string, status *string) ([]*restModel.APITest, error) {
+func (r *queryResolver) PatchTasks(ctx context.Context, patchID string, sortBy *TaskSortCategory, sortDirection *SortDirection, page *int, limit *int, searchInput *string, statuses []*string) ([]*TaskResult, error) {
+	sorter := ""
+	if sortBy != nil {
+		switch *sortBy {
+		case TaskSortCategoryStatus:
+			sorter = task.StatusKey
+			break
+		case TaskSortCategoryName:
+			sorter = task.DisplayNameKey
+			break
+		case TaskSortCategoryBaseStatus:
+			// base status is not a field on the task db model; therefore sorting by base status
+			// cannot be done in the mongo query. sorting by base status is done in the resolver.
+			break
+		case TaskSortCategoryVariant:
+			sorter = task.BuildVariantKey
+			break
+		default:
+			break
+		}
+	}
+	sortDir := 1
+	if *sortDirection == SortDirectionDesc {
+		sortDir = -1
+	}
+	pageParam := 0
+	if page != nil {
+		pageParam = *page
+	}
+	limitParam := 0
+	if limit != nil {
+		limitParam = *limit
+	}
+	statusesParam := []string{}
+	if statuses != nil {
+		for _, status := range statuses {
+			statusesParam = append(statusesParam, *status)
+		}
+	}
+	taskOrVariantName := ""
+	if searchInput != nil {
+		taskOrVariantName = *searchInput
+	}
+
+	tasks, err := r.sc.FindTaskResultsByVersion(patchID, taskOrVariantName, sorter, statusesParam, sortDir, pageParam, limitParam)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting patch tasks for %s: %s", patchID, err.Error()))
+	}
+	if tasks == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("No tasks found for %s", patchID))
+	}
+
+	baseTaskStatuses, err := GetBaseTaskStatusesFromPatchID(r, patchID)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting base task statuses for %s: %s", patchID, err.Error()))
+	}
+
+	var taskResults []*TaskResult
+	for _, task := range tasks {
+		t := TaskResult{
+			ID:           task.Id,
+			DisplayName:  task.DisplayName,
+			Version:      task.Version,
+			Status:       task.Status,
+			BuildVariant: task.BuildVariant,
+			BaseStatus:   baseTaskStatuses[task.BuildVariant][task.DisplayName],
+		}
+		taskResults = append(taskResults, &t)
+	}
+	if *sortBy == TaskSortCategoryBaseStatus {
```

Comment:
We should check if sortBy is nil before evaluating this condition

Reply:
sortBy will never be nil because it has a default value of `STATUS` per the resolver signature in the schema.